helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -250,6 +250,12 @@ private boolean computeResourceBestPossibleState(ClusterEvent event, ResourceCon
         // The next release will support rebalancers that compute the mapping from start to finish
         partitionStateAssignment = mappingCalculator
             .computeBestPossiblePartitionState(cache, idealState, resource, currentStateOutput);
+
+        if (partitionStateAssignment == null) {
+          LogUtil.logError(logger, _eventId, "PartitionStateAssignment is null.");
```

Comment:
Let's use warning instead of error, since it is not a critical failure. Also please add the resource name in the message, otherwise, we may not know which resource has this problem.

Reply:
@dasahcc done